src/wallet/wallet2.cpp

```diff
@@ -6389,6 +6754,22 @@ void wallet2::store_to(const std::string &path, const epee::wipeable_string &pas
     // store should only exist if the MMS is really active
     m_message_store.write_to_file(get_multisig_wallet_state(), m_mms_file);
   }
+
+  if (m_background_sync_type == BackgroundSyncCustomPassword && !m_background_syncing && !m_is_background_wallet)
+  {
+    // Update the background wallet cache when we store the main wallet cache
+    // Note: if background syncing when this is called, it means the background
+    // wallet is open and was already stored above
+    try
+    {
+      THROW_WALLET_EXCEPTION_IF(!m_custom_background_key, error::wallet_internal_error, "Custom background key not set");
```

Comment:
Nit:

Instead of [using exceptions to control program flow](https://isocpp.org/wiki/faq/exceptions#why-not-exceptions), consider an alternative, such as a a `TryGet` method.

Pseudocode:

```
BackgroundKey backgroundKey;
if (!TryGet(backgroundKey))
{
    MERROR("Failed to update background cache, custom background key not set");
    return;
}
```

```

if (TryGet(

Reply:
This alternative is 4 more lines of code versus 1 in this particular spot, I'm going to keep it as is and I don't think it's worth arguing over

Reply:
Cool